.github/workflows/build_steps.yml

```diff
@@ -207,8 +229,24 @@ jobs:
       test_dirs: ${{env.test_dirs}}
       cibw_build: ${{env.CIBW_BUILD}}
 
+  stop-ec2-runner:
+    needs: [start_ec2_runner, compile]
+    if: |
+      always() &&
+      inputs.distro == 'ec2'
+    uses: ./.github/workflows/ec2_runner_jobs.yml
+    secrets: inherit
+    with:
+      job_type: stop
+      label: ${{ needs.start_ec2_runner.outputs.label }}
+      ec2-instance-id: ${{ needs.start_ec2_runner.outputs.ec2-instance-id }}
+
   python_tests:
-    if: inputs.job_type == 'follower'
+    if: |
+      always() &&
+      !failure() &&
+      !cancelled() &&
```

Comment:
Use `continue-on-error: true` on the previous step instead of these.

You may also want to set an auto-shutdown on all EC2 instances we start, in case any kind of error causes an instance not to get killed.

Reply:
This is addressing a limitation of GH Actions. Essentially because compile depends on the stert_ec2 job, if that step is skipped, GHA skips **all** downstream jobs by default. So we need to add these to explicitly enable them, even if the ster_ec2 job is skipped, e.g. on Windows.

The step for stopping/deleting the ec2 runners is always executed, but I will look into, if there is a way to add some timeout just in case.

Reply:
What I meant was if you put `continue-on-error: true` on a step that can be skipped, it will not skip subsequent steps anymore, which is much clearer in the intention compared to this `if` condition.

Reply:
Looks like continue-on-error is not supported on a job with a uses clause